File: leetcode/2161.js
/**
 * @param {number[]} nums
 * @param {number} pivot
 * @return {number[]}
 */
var pivotArray = function(nums, pivot) {
    let ls = [];
    let es = [];
    let ms = [];

    nums.forEach(n => {
        ((n < pivot) ? ls : ((n === pivot) ? es : ms)).push(n)
    });

    return ls.concat(es).concat(ms);
};

File: leetcode/2161.scala
object Solution {
    def pivotArray(nums: Array[Int], pivot: Int): Array[Int] = {
        val (part1, part2) = nums partition { _ < pivot }
        val (part2a, part2b) = part2 partition { _ == pivot }
        part1 :++ part2a :++ part2b
    }
}
